Index: ../UncolDatabaseSample/app/src/main/java/jp/ac/it_college/std/s22021/uncol_database_sample/AddActivity.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/../UncolDatabaseSample/app/src/main/java/jp/ac/it_college/std/s22021/uncol_database_sample/AddActivity.kt b/../UncolDatabaseSample/app/src/main/java/jp/ac/it_college/std/s22021/uncol_database_sample/AddActivity.kt
new file mode 100644
--- /dev/null	
+++ b/../UncolDatabaseSample/app/src/main/java/jp/ac/it_college/std/s22021/uncol_database_sample/AddActivity.kt	
@@ -0,0 +1,35 @@
+package jp.ac.it_college.std.s22021.uncol_database_sample
+
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import jp.ac.it_college.std.s22021.uncol_database_sample.databinding.ActivityAddBinding
+
+class AddActivity : AppCompatActivity() {
+    privat lateinit
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityAddBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        binding.saveButton.setOnClickListener(::onSave)
+
+        helper = DatabaseHelper(this)
+    }
+
+    private fun onSave(v: View): Unit {
+        // データの保存処理をする。
+        helper.writableDatabase.let { ad ->
+            val insertSQL = """
+                INSERT INTO memo (name, content)
+                VALUES (?, ?);
+            """.trimIndent()
+            db.compileStatement(insertSQL).use { stmt ->
+                stmt.bindString(1, binding.name.text.toString())
+                stmt.bindString(2, binding.content.text.toString())
+                stmt.executeInsert()
+            }
+        }
+        // 保存が終わったら元の画面（一覧画面）に戻る。
+        finish()
+    }
+}
\ No newline at end of file
Index: ../UncolDatabaseSample/app/src/main/res/layout/activity_add.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/../UncolDatabaseSample/app/src/main/res/layout/activity_add.xml b/../UncolDatabaseSample/app/src/main/res/layout/activity_add.xml
new file mode 100644
--- /dev/null	
+++ b/../UncolDatabaseSample/app/src/main/res/layout/activity_add.xml	
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context=".AddActivity">
+
+    <TextView
+        android:id="@+id/titleLabel"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="@string/add_data"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent" />
+
+</androidx.constraintlayout.widget.ConstraintLayout>
\ No newline at end of file
Index: ../UncolDatabaseSample/app/src/main/java/jp/ac/it_college/std/s22021/uncol_database_sample/DatabaseHelper.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/../UncolDatabaseSample/app/src/main/java/jp/ac/it_college/std/s22021/uncol_database_sample/DatabaseHelper.kt b/../UncolDatabaseSample/app/src/main/java/jp/ac/it_college/std/s22021/uncol_database_sample/DatabaseHelper.kt
new file mode 100644
--- /dev/null	
+++ b/../UncolDatabaseSample/app/src/main/java/jp/ac/it_college/std/s22021/uncol_database_sample/DatabaseHelper.kt	
@@ -0,0 +1,23 @@
+package jp.ac.it_college.std.s22021.uncol_database_sample
+
+import android.content.Context
+import android.database.sqlite.SQLiteDatabase
+import android. database.sqlite.SQLiteOpenHelper
+
+private  const val DB_NAME = "memo_database.db"
+private  const val DB_VERSION = 1
+
+class DatabaseHelper(context: Context) : SQLiteOpenHelper(
+    context, DB_NAME, null, DB_VERSION
+) {
+    override fun onCreate(db: SQLiteDatabase?) {
+        // テーブルを作る
+        val createTable = """
+            CREATE TABLE memo (
+                _id INTEGER PRIMARY KEY AUTOINCREMENT,
+                name TEXT,
+                content TEXT
+            );
+        """.trimIndent()
+    }
+}
\ No newline at end of file
